TravelerApp/TravelerAppCore/Controller/Search.cs

```diff
@@ -25,36 +25,35 @@ public static List<Hotel> ByLocalisation(string address)
             return hotelLocalisation;
         }
 
-        //public static List<int> HotelsByName(string searchedName)
-        //{
-        //    List<int> foundHotels = new List<int>();
-        //    int exactMatch = 0;
-        //    //check if a search name has at least one character in it
-        //    // search for an exact match first- if found we exit a method after finding one - I assumed hotel names are distinct
-        //    for (int i = 0; i < HotelService.Data.Count; i++)
-        //    {
-        //        if (HotelService.Data[i].HotelInfo.Name.ToUpper().Equals(searchedName.ToUpper()))
-        //        {
-        //            exactMatch++;
-        //            foundHotels.Add(i);
-        //        }
-        //    }
-        //    if (exactMatch != 0)
-        //    {
-        //        return foundHotels;
-        //    }
-        //    else
-        //    //if no exact match has been found- we look for partial match and return all the hotels found
-        //    {
-        //        for (int i = 0; i < HotelService.Data.Count; i++)
-        //        {
-        //            if (HotelService.Data[i].HotelInfo.Name.ToUpper().Contains(searchedName.ToUpper()))
-        //            {
-        //                foundHotels.Add(i);
-        //            }
-        //        }
-        //        return foundHotels;
-        //    }
-        //}
+        public static List<Hotel> ByName(string name)
```

Comment:
jeśli masz czas to możesz spróbować w ramach treningu zamienić to wyszukiwanie z użyciem linq